woocommerce/woocommerce-android
# Issue/6936 dismiss card reader upsell banner

Closes: #6935 , #6936 


### Description

This PR implements business logic for "Remind me later" and "Don't show again" CTAs in the banner dismiss the alert dialog in the payment selection screen.

For `Remind me later`, we show the upsell card reader banner again after 14 days after the last dismissal.

For `Don't show again`, we never show the upsell card reader banner again unless the merchant clears the app data.


### Note to reviewer
[This PR](https://github.com/woocommerce/woocommerce-android/pull/6981) does some refactoring of the business logic of both `Remind me later` and `Don't show again`. I would suggest referring to that PR to better understand how the code evolves and just to keep in mind that the business logic in this PR gets refactored in the later PR.

### Testing instructions

1. Navigate to any orders eligible for IPP
2. Try to dismiss the banner
3. Ensure you see an alert dialog asking to either remind later or never show again
4. Play around by clicking both the options and make sure the app behaves as expected

When `Remind me later` clicked:
For easier testing, modify the code in `SelectPaymentMethodViewModel` lineno: `322` method `isLastDialogDismissedMoreThan14DaysAgo` like this
```
val timeDifference = currentTimeInMillis - getCardReaderUpsellBannerLastDismissed()
        return TimeUnit.MILLISECONDS.toSeconds(timeDifference) > 10
```
The above piece of code basically says the threshold for displaying the banner again after the last dismissal is 5 seconds. After 5 seconds, the banner starts appearing again on the payments screen.


When `Don't show again` clicked:
The banner must not be shown again unless you clear the app data.

### Images/gif

**Don't Show Again**

https://user-images.githubusercontent.com/1331230/179734716-dc79983e-6ea1-4a62-8ec2-915a0dd27760.mp4


**Remind Me Later with threshold set as 5 seconds for demonstration**


https://user-images.githubusercontent.com/1331230/179734919-b1b2c143-7665-4c2d-9180-398b4c1a0ecf.mp4
